hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncNonMetaRegionLocator.java

```diff
@@ -91,6 +95,9 @@ class AsyncNonMetaRegionLocator {
 
   private final ConcurrentMap<TableName, TableCache> cache = new ConcurrentHashMap<>();
 
+  // A chore service to invalidate table cache which table is not exist or disabled.
+  private ChoreService metaCacheInvalidateChoreService;
```

Comment:
We have a timer in AsyncConnectionImpl, better reuse that thread to do this.

Reply:
We scan schedule a new task at the end of the current task. And yes, you can not do block waiting in the timer task, but since we are using AsyncTable here, you can register a callback in the return CompletableFuture to avoid blocking the timer thread.